elastic/kibana
# [Mappings Editor] Disable _source field in serverless

Closes https://github.com/elastic/kibana/issues/181555

## Summary

This PR disables the `_source` field in the Mappings editor's advanced options when in serverless or when the `_source.mode` property is set.

**How to test:**
1. Start a serverless project.
2. Go to Index Management and start creating an index template or a component template.
3. In the Mappings step, go to the "Advanced options" tab.
4. Verify that the `_source` field is not displayed and that creating a template doesn't add this property to the Es request.
5. Verify that, in stateful Kibana, the `_source` field still exists and works as expected.
6. In both stateful and serverless, verify that when you create an index template with the `_source.mode` property set (example Es request below), the whole _source field section is hidden.
```
PUT _index_template/my_template
{
  "index_patterns": ["foo*"],
  "template": {
    "mappings": {
      "_source": {
        "mode": "synthetic"
      },
      "properties": {
        "host_name": {
          "type": "keyword"
        },
        "created_at": {
          "type": "date",
          "format": "EEE MMM dd HH:mm:ss Z yyyy"
        }
      }
    }
  }
}

```
